tools/coq_makefile.ml

```diff
@@ -256,18 +252,26 @@ let generate_conf_files oc
   { v_files; mli_files; mlg_files; ml_files; mllib_files; mlpack_files; meta_file }
 =
   let module S = String in
-  let map = map_sourced_list in
+  let map_wildcard s = (* allow for generated source files not initially present *)
+    if try ignore (S.index_from s 0 '*'); true with Not_found -> false then
+      Printf.sprintf "$(wildcard %s)" s
+      else s
+  in
+  let fout varname values =
+    let rhs = List.map map_wildcard (map_sourced_list quote values) in
+    fprintf oc "COQMF_%s = %s\n" varname (S.concat " " rhs)
+  in
   section oc "Project files.";
-  fprintf oc "COQMF_VFILES = %s\n"      (S.concat " " (map quote v_files));
-  fprintf oc "COQMF_MLIFILES = %s\n"    (S.concat " " (map quote mli_files));
-  fprintf oc "COQMF_MLFILES = %s\n"     (S.concat " " (map quote ml_files));
-  fprintf oc "COQMF_MLGFILES = %s\n"    (S.concat " " (map quote mlg_files));
-  fprintf oc "COQMF_MLPACKFILES = %s\n" (S.concat " " (map quote mlpack_files));
-  fprintf oc "COQMF_MLLIBFILES = %s\n"  (S.concat " " (map quote mllib_files));
+  fout "VFILES" v_files;
+  fout "MLIFILES" mli_files;
+  fout "MLFILES" ml_files;
+  fout "MLGFILES" mlg_files;
+  fout "MLPACKFILES" mlpack_files;
+  fout "MLLIBFILES" mllib_files;
   fprintf oc "COQMF_METAFILE = %s\n"  (match meta_file with Present x -> x | _ -> "");
   let cmdline_vfiles = filter_cmdline v_files in
-  fprintf oc "COQMF_CMDLINE_VFILES = %s\n" (S.concat " " (List.map quote cmdline_vfiles));
-;;
+  let rhs = List.map (fun s -> map_wildcard (quote s)) cmdline_vfiles in
+  fprintf oc "COQMF_CMDLINE_VFILES = $(wildcard %s)\n" (S.concat " " rhs);
```

Comment:
This seems wrong.
Since `rhs` was mapped via `map_wildcard` you already have `$(wildcard xxx)` around each item, so you should not add one more. This also makes it possible to use `fout` as you do for the other variables.

Reply:
Good catch on the repetition.  However `fout` has `(map_sourced_list quote values)` which won't work for line 273, right?

Reply:
Yes you are right.